strawberryfields/backends/states.py

```diff
@@ -621,6 +622,23 @@ def wigner(self, mode, xvec, pvec):
 
         return W / (self._hbar)
 
+    def p_quad_values(self, mode, xvec, pvec):
+
+        W = BaseFockState.wigner(mode, xvec, pvec)
+        y = []
+        for i in range(0, len(xvec)):
+            res = simps([W[i][k] for k in range(0, len(xvec))], xvec)
```

Comment:
Can this be changed to the following?
```suggestion
            res = simps(W[i, :len(xvec)], xvec)
```

This _should_ work since `W` is a NumPy array so supports fancier indexing (and should be faster than a list comprehension)